### PR TITLE
Use FairCMakeModules for find_package2

### PR DESCRIPTION
**Pull request by ChristianTackeGSI**
* Make cmake module installation more granular (only install specific subdirs of `cmake/`)
* Optionally use FairCMakeModules for some modules. Fallback is in `cmake/fallback/`
* Extract `find_package` into `FairFindPackage2.cmake` (added to fallback directory)

---

Checklist:

* [X] Rebased against `dev` branch
* [X] My name is in the resp. CONTRIBUTORS/AUTHORS file
* [X] Followed [the seven rules of great commit messages](https://chris.beams.io/posts/git-commit/#seven-rules)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -98,6 +98,12 @@ If(NOT _HAS_CXX11_FLAG)
 EndIf()
 
 # Load some basic macros which are needed later on
+find_package(FairCMakeModules 0.1 CONFIG)
```
Comment (dennisklein) [inline CMakeLists.txt:101]: Why is the `CONFIG` needed?

File: CMakeLists.txt [hunk 2]
```diff
@@ -98,6 +98,12 @@ If(NOT _HAS_CXX11_FLAG)
...
+if(NOT FairCMakeModules_FOUND)
+  message(WARNING "Please highly consider installing FairCMakeModules!")
```
Comment (dennisklein) [inline CMakeLists.txt:103]:
```suggestion
  message(WARNING "Could not find FairCMakeModules, falling back to local copy. It is recommended to install https://github.com/FairRootGroup/FairCMakeModules!")
```

Comment (dennisklein) [reply CMakeLists.txt:103]: Maybe also consider making it a `STATUS` message, `WARNING` is quite "loud".

File: CMakeLists.txt (hunk 1 above)
Comment (ChristianTackeGSI) [reply CMakeLists.txt:101]: I found the verbosity about find-modules annoying. As we now have a dedicated warning (see below), a QUIET is probbaly better?

File: CMakeLists.txt (hunk 2 above)
Comment (ChristianTackeGSI) [reply CMakeLists.txt:103]: Will take the better message!

Well, I would like to remove the fallback in 19.2. So I want to be loud in the 19.0 release.

**Review by dennisklein — APPROVED** [review]

**Pull request merged**